### ISSUE TITLE
Model.remove return unnecessary field "connection"

### ISSUE THREAD

**Issue by Fonger** [Automattic/mongoose#4013]
The promise resolved with following value:

What we need is `result`.
The `connection` should not be returned or make it not enumerable as it contains private information such as `ssl`, `key`, `host`, `port` etc

``` js
   { result: { ok: 1, n: 0 },
     connection:
      EventEmitter {
        domain: null,
        _events: [Object],
        _eventsCount: 4,
        _maxListeners: undefined,
        options: [Object],
        id: 1,
        logger: [Object],
        bson: {},
        tag: undefined,
        messageHandler: [Function],
        maxBsonMessageSize: 67108864,
        port: 12345,
        host: 'xxxxx.xxx.net',
        keepAlive: true,
        keepAliveInitialDelay: 0,
        noDelay: true,
        connectionTimeout: 0,
        socketTimeout: 0,
        destroyed: false,
        domainSocket: false,
        singleBufferSerializtion: true,
        serializationFunction: 'toBinUnified',
        ca: <Buffer 2d 2d 2d 2d 7d 42 45 47 49 4e 20 43 45 52 54 42 46 49 43 41 54 45 2d 2d 2d 2d 2d 0a 4d 49 49 45 53 54 43 43 41 7a 57 67 41 77 49 42 41 67 49 4a 42 50 ... >,
        cert: <Buffer 2d 2d 2d 2d 7d 42 45 47 49 4e 20 43 45 52 54 49 46 49 43 41 54 45 2d 2d 2d 2d 2d 0a 4d 49 49 45 50 7a 43 43 41 79 65 67 41 77 49 42 41 63 49 4a 41 4f ... >,
        key: <Buffer 2d 2d 2d 2d 2d 22 45 47 49 4e 20 50 52 49 56 41 54 45 20 4b 45 59 2d 2d 2d 2d 2d 0a 4d 49 49 45 77 41 49 42 41 44 41 4e 42 67 6b 71 68 6b 69 47 39 77 ... >,
        passphrase: null,
        ssl: true,
        rejectUnauthorized: true,
        checkServerIdentity: true,
        responseOptions: [Object],
        flushing: false,
        queue: [],
        connection: [Object],
        writeStream: null,
        hashedName: '45416958fxxxxxxxxxxxxxxxea140a3a9517c63f',
        buffer: null,
        sizeOfMessage: 0,
        bytesRead: 0,
        stubBuffer: null } } }
```

code to reproduce:

``` js
Review.remove({ _book: bookId }, { multi: true }).then(result => {
    console.log(result);
});
```

**Comment by vkarpov15** [comment]
Would be a backwards breaking change, but IMO it's a good one. In the meantime, `.then(res => res.result)` should work

**Comment by vkarpov15** [comment]
Fixed in 5.0 branch
